opentelemetry-sdk/src/metrics/mod.rs

```diff
@@ -925,6 +925,62 @@ mod tests {
         assert_eq!(data_point1.value, 6);
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn counter_aggregation_attribute_ordering() {
```

Comment:
added this to investigate some issue, that was false alarm. Keeping this test for now. It is not that useful as-is, but can be augmented in subsequent PRs to make sure we cover all scenarios via testing.